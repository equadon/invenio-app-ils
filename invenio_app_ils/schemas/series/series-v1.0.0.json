{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "Series",
  "type": "object",
  "required": ["$schema", "pid", "title", "mode_of_issuance"],
  "properties": {
    "$schema": {
      "type": "string"
    },
    "abstract": {
      "type": "string",
      "description": ""
    },
    "access_urls": {
      "items": {
        "additionalProperties": false,
        "properties": {
          "access": {
            "minLength": 1,
            "type": "string"
          },
          "description": {
            "minLength": 1,
            "type": "string"
          },
          "value": {
            "format": "uri",
            "minLength": 1,
            "type": "string"
          }
        },
        "required": ["value"],
        "title": "URL of external resource",
        "type": "object"
      },
      "minItems": 1,
      "type": "array",
      "uniqueItems": true
    },
    "alternative_titles": {
      "description": "Alternative titles of the series",
      "items": [
        {
          "additionalProperties": false,
          "properties": {
            "value": {
              "minLength": 1,
              "type": "string"
            },
            "type": {
              "type": "string",
              "title": "Type of alternative title (Vocabulary)"
            }
          },
          "required": ["value"],
          "type": "object"
        }
      ],
      "minItems": 1,
      "type": "array",
      "uniqueItems": true
    },
    "authors": {
      "items": {
        "type": "string"
      },
      "type": "array",
      "description": ""
    },
    "edition": {
      "type": "string",
      "description": "Edition"
    },
    "identifiers": {
      "items": {
        "properties": {
          "material": {
            "type": "string",
            "title": "Material to which the identifiers refers to."
          },
          "scheme": {
            "title": "Scheme of the identifier (Vocabulary)",
            "type": "string"
          },
          "value": {
            "title": "Value of the identifier",
            "type": "string"
          }
        },
        "required": ["value", "scheme"]
      },
      "title": "List of Identifiers of the record",
      "type": "array"
    },
    "internal_notes": {
      "title": "Internal notes",
      "type": "string"
    },
    "languages": {
      "items": {
        "type": "string"
      },
      "type": "array",
      "title": "List of languages of the series (Vocabulary)",
      "description": ""
    },
    "mode_of_issuance": {
      "title": "Mode of issuance",
      "type": "string",
      "enum": ["MULTIPART_MONOGRAPH", "SERIAL"],
      "description": ""
    },
    "notes": {
      "title": "Note for the public",
      "type": "string"
    },
    "publisher": {
      "title": "Publisher name",
      "type": "string"
    },
    "relations_metadata": {
      "type": "object",
      "title": "Metadata to describe relations",
      "properties": {
        "multipart_monograph": {
          "type": "array",
          "items": {
            "type": "object",
            "properties": {
              "pid": {
                "type": "string"
              },
              "pid_type": {
                "type": "string"
              },
              "volume": {
                "type": "string"
              }
            }
          }
        },
        "serial": {
          "type": "array",
          "items": {
            "type": "object",
            "properties": {
              "pid": {
                "type": "string"
              },
              "pid_type": {
                "type": "string"
              },
              "volume": {
                "type": "string"
              }
            }
          }
        },
        "other": {
          "type": "array",
          "items": {
            "type": "object",
            "properties": {
              "pid": {
                "type": "string"
              },
              "pid_type": {
                "type": "string"
              },
              "note": {
                "type": "string"
              }
            }
          }
        }
      }
    },
    "relations": {
      "type": "object",
      "properties": {
        "$ref": {
          "type": "string"
        }
      }
    },
    "pid": {
      "title": "Series PID",
      "type": "string"
    },
    "title": {
      "type": "string"
    },
    "urls": {
      "items": {
        "additionalProperties": false,
        "properties": {
          "description": {
            "minLength": 1,
            "type": "string"
          },
          "value": {
            "format": "uri",
            "minLength": 1,
            "type": "string"
          }
        },
        "required": ["value"],
        "title": "URL of external resource",
        "type": "object"
      },
      "minItems": 1,
      "type": "array",
      "uniqueItems": true
    }
  }
}
