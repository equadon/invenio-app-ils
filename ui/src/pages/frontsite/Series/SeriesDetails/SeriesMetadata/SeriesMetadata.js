import React, { Component } from 'react';
import { Container, Responsive } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import {
  SeriesMetadataAccordion,
  SeriesMetadataTabs,
} from '@pages/frontsite/components/Series';

export default class SeriesMetadata extends Component {
  render() {
    const { series } = this.props;
    return (
      <Container className="document-metadata" data-test={series.metadata.pid}>
        <Responsive minWidth={Responsive.onlyTablet.minWidth}>
          <SeriesMetadataTabs />
        </Responsive>
        <Responsive {...Responsive.onlyMobile}>
          <SeriesMetadataAccordion />
        </Responsive>
      </Container>
    );
  }
}

SeriesMetadata.propTypes = {
  series: PropTypes.object.isRequired,
};
