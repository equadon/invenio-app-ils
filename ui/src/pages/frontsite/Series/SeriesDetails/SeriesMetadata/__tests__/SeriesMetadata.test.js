import React from 'react';
import { mount } from 'enzyme';
import * as testData from '@testData/series.json';
import SeriesMetadata from '../SeriesMetadata';

jest.mock('@pages/frontsite/components/Series', () => {
  return {
    SeriesMetadataAccordion: () => null,
    SeriesMetadataTabs: () => null,
  };
});

describe('SeriesMetadata tests', () => {
  let component;

  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const seriesDetails = {
    id: 71,
    created: '2019-07-08T10:44:02.366+02:00',
    updated: '2019-07-08T10:44:18.354+02:00',
    links: { self: 'https://127.0.0.1:5000/api/series/71' },
    metadata: {
      $schema: 'https://127.0.0.1:5000/schemas/series/series-v1.0.0.json',
      ...testData[0],
      pid: '71',
    },
  };

  it('should render the series correctly', () => {
    component = mount(<SeriesMetadata series={seriesDetails} />);
    expect(component).toMatchSnapshot();

    const rows = component
      .find('SeriesMetadata')
      .find('Container')
      .filterWhere(
        element => element.prop('data-test') === seriesDetails.metadata.pid
      );
    expect(rows).toHaveLength(1);
  });
});
