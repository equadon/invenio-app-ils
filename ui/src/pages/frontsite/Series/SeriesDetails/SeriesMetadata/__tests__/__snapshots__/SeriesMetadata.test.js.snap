// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesMetadata tests should render the series correctly 1`] = `
<SeriesMetadata
  series={
    Object {
      "created": "2019-07-08T10:44:02.366+02:00",
      "id": 71,
      "links": Object {
        "self": "https://127.0.0.1:5000/api/series/71",
      },
      "metadata": Object {
        "$schema": "https://127.0.0.1:5000/schemas/series/series-v1.0.0.json",
        "abstract": "This is a multipart monograph",
        "authors": Array [
          "Einstein, Albert",
          "Stachel, John J et al.",
        ],
        "mode_of_issuance": "MULTIPART_MONOGRAPH",
        "pid": "71",
        "title": "The Gulf: The Making of An American Sea",
      },
      "updated": "2019-07-08T10:44:18.354+02:00",
    }
  }
>
  <Container
    className="document-metadata"
    data-test="71"
  >
    <div
      className="ui container document-metadata"
      data-test="71"
    >
      <Responsive
        getWidth={[Function]}
        minWidth={768}
      >
        <div>
          <SeriesMetadataTabs />
        </div>
      </Responsive>
      <Responsive
        getWidth={[Function]}
        maxWidth={767}
        minWidth={320}
      />
    </div>
  </Container>
</SeriesMetadata>
`;
